### DISCUSSION TITLE
Export `getRuleValue()` as a public API

### DISCUSSION THREAD

**Discussion by jonkoops** [react-hook-form/react-hook-form#12181] [Q&A]
**Is your feature request related to a problem? Please describe.**
I often need to extract the value of a validation rule to wire up some validation to my component library.

**Describe the solution you'd like**
I'd like to see [`getRuleValue()`](https://github.com/react-hook-form/react-hook-form/blob/ea0f3ed86457691f79987a703ae8d50b9e16e2ad/src/logic/getRuleValue.ts#L10-L21) exported as part of the public API so it can be re-used.

**Describe alternatives you've considered**
Currently rolling my own implementation, which is fine, but why duplicate code.

**Additional context**

**Comment by bluebill1049** [comment]
you can actually just get the source code and put into your own app if you need it.

**Reply by jonkoops** [reply]
Yes, I could do that, but then it would also make sense to have this as a public API instead no? I don't think it is an uncommon use-case to want to get the value provided to a validation rule, if you need to wire up any sort of component system you'll likely run into this.
